### PR TITLE
doc: guides: app: Rework for multi-OS support

### PR DESCRIPTION
**Pull request by carlescufi**
Rework the Application Development Primer so that it describes the
differences between UNIX and Windows builds and uses Ninja by default,
in order to be compatible with all of them.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

### CODE REVIEW

File: doc/application/application.rst [hunk 1]
```diff
@@ -131,11 +132,11 @@ shown above in :file:`CMakeLists.txt`. The following important variables
 configure the Zephyr build system:
 
 * :makevar:`ZEPHYR_BASE`: Sets the path to the Zephyr base directory.  This is
-  usually an environment variable set by the :file:`zephyr-env.sh` script, as
-  you learned when getting started with Zephyr in
-  :ref:`getting_started_run_sample`. You can also select a specific Zephyr base
-  directory by replacing ``$ENV{ZEPHYR_BASE}`` with the specific base you'd
-  like to use instead.
+  usually an environment variable set by the :file:`zephyr-env.sh` script on
+  Linux/macOS or manually on Windows,, as you learned when getting started
```
Comment (dbkinder) [inline doc/application/application.rst]: delete the extra comma after ``Windows,,``

**Review by dbkinder — CHANGES_REQUESTED** [review]
one typo, otherwise LGTM

**Comment by carlescufi** [comment]
@dbkinder Fixed, but https://github.com/zephyrproject-rtos/zephyr/pull/5708 needs to be merged first in order for me to rebase and for CI to pass

**Review by dbkinder — APPROVED** [review]

File: doc/application/application.rst [hunk 2]
```diff
@@ -131,11 +132,11 @@ shown above in :file:`CMakeLists.txt`. The following important variables
...
 
 * :makevar:`ZEPHYR_BASE`: Sets the path to the Zephyr base directory.  This is
-  usually an environment variable set by the :file:`zephyr-env.sh` script, as
-  you learned when getting started with Zephyr in
-  :ref:`getting_started_run_sample`. You can also select a specific Zephyr base
-  directory by replacing ``$ENV{ZEPHYR_BASE}`` with the specific base you'd
-  like to use instead.
+  usually an environment variable set by the :file:`zephyr-env.sh` script on
+  Linux/macOS or manually on Windows, as you learned when getting started
+  with Zephyr in :ref:`getting_started_run_sample`. You can also select a
```
Comment (mbolivar) [inline doc/application/application.rst]: This last sentence ("You can also select ... use instead") is redundant on Windows now. Maybe reword to:

```
(You can also set :makevar:`ZEPHYR_ENV` explicitly on Unix environments if you want to.)
```

File: doc/application/application.rst [hunk 3]
```diff
@@ -157,23 +158,25 @@ The Zephyr build system compiles and links all components of an application
 into a single application image that can be run on simulated hardware or real
 hardware.
 
+As described in `_getting_started_cmake`, on Linux and macOS you can choose
```
Comment (mbolivar) [inline doc/application/application.rst]: This would be a bit better as:

```
:ref:`_getting_started_cmake`
```

Just in case the content gets moved around.

Reference:

> Using ref is advised over standard reStructuredText links to sections (like `Section title`_) because it works across files, when section headings are changed, and for all builders that support cross-references.

http://www.sphinx-doc.org/en/stable/markup/inline.html#cross-referencing-arbitrary-locations

File: doc/application/application.rst [hunk 4]
```diff
@@ -768,8 +767,6 @@ preferred, since it correctly handles dependencies between options.
 #. Set kernel configuration values using the following
    key commands:
 
-   * Use the arrow keys to navigate within any menu or list.
```
Comment (mbolivar) [inline doc/application/application.rst]: Is this no longer supported? I think this should be left here.

**Review by mbolivar — COMMENTED** [review]
Just a few things

File: doc/application/application.rst (hunk 2 above)
Comment (carlescufi) [reply doc/application/application.rst]: Fixed

File: doc/application/application.rst (hunk 4 above)
Comment (carlescufi) [reply doc/application/application.rst]: Oops, this was a mistake, fixed, thanks!

**Review by mbolivar — APPROVED** [review]
LGTM
